reference-implementation/lib/writable-stream.js

```diff
@@ -332,15 +315,16 @@ function WritableStreamHandleAbortRequestIfPending(stream) {
     return;
   }
 
-  WritableStreamFinishAbort(stream);
-
   const abortRequest = stream._pendingAbortRequest;
   stream._pendingAbortRequest = undefined;
   const promise = WritableStreamDefaultControllerAbort(stream._writableStreamController, abortRequest._reason);
   promise.then(
     abortRequest._resolve,
     abortRequest._reject
   );
+  stream._backpressure = false;
```

Comment:
This feels a little hacky. It is necessary to make sure we go down the "reset the ready promise" path inside WritableStreamError, instead of the "reject the ready promise" path. Is there a better way?

The reason it doesn't feel terrible is that yeah, we're now processing an abort request, so there's not really any backpressure I guess...